src/QMCWaveFunctions/ElectronGas/FreeOrbital.h

```diff
@@ -24,9 +24,11 @@ namespace qmcplusplus
 class FreeOrbital : public SPOSet
 {
 public:
-  FreeOrbital(const std::vector<PosType>& kpts_cart);
+  FreeOrbital(const std::string& my_name, const std::vector<PosType>& kpts_cart);
   ~FreeOrbital();
 
+  std::string getClassName() const override { return "FreeOrbital"; }
```

Comment:
If this is just used for error reporting/debugging, which is what I see, its fine if we need runtime reflection I don't think we should do it via string matching.

Reply:
Agreed. If we need to make selection based on derived class types, we should do enum not strings. It is only intended for printing.